File: src/main/java/net/sf/rails/game/model/PurseMoneyModel.java
package net.sf.rails.game.model;

import net.sf.rails.game.financial.Bank;
import net.sf.rails.game.financial.RailsMoneyOwner;
import net.sf.rails.game.state.BooleanState;
import net.sf.rails.game.state.Currency;
import net.sf.rails.game.state.Purse;

/**
 * A MoneyModel with a wallet inside
 */
public class PurseMoneyModel extends MoneyModel {

    private final Purse purse;

    private final BooleanState initialised;

    private PurseMoneyModel(RailsMoneyOwner parent, String id, Boolean init, Currency currency) {
        super(parent, id, currency);

        this.purse = Purse.create(parent, "purse", currency);
        this.purse.addModel(this);

        this.initialised = new BooleanState(this, "initialised", init);
    }

    public static PurseMoneyModel create(Bank parent, String id, Boolean init, Currency currency) {
        return new PurseMoneyModel(parent, id, init, currency);
    }

    public static PurseMoneyModel create(RailsMoneyOwner parent, String id, Boolean init) {
        Currency currency = parent.getRoot().getBank().getCurrency();
        return new PurseMoneyModel(parent, id, init, currency);
    }

    @Override
    public RailsMoneyOwner getParent() {
        return (RailsMoneyOwner) super.getParent();
    }

    public Purse getPurse() {
        return purse;
    }

    // MoneyModel abstracts
    @Override
    public int value() {
        return purse.value();
    }

    @Override
    public boolean initialised() {
        return initialised.value();
    }

}

File: dotnet/RailsLib.Net/Net/Game/Model/PurseMoneyModel.cs
using GameLib.Net.Game.Financial;
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Text;

/**
 * A MoneyModel with a wallet inside

 */
namespace GameLib.Net.Game.Model
{
    public class PurseMoneyModel : MoneyModel
    {
        private Purse purse;

        private BooleanState initialized;

        private PurseMoneyModel(IRailsMoneyOwner parent, string id, Boolean init, Currency currency) : base(parent, id, currency)
        {
            purse = Purse.Create(parent, "purse", currency);
            purse.AddModel(this);
            initialized = BooleanState.Create(this, "initialized", init);
        }

        public static PurseMoneyModel Create(Bank parent, string id, bool init, Currency currency)
        {
            return new PurseMoneyModel(parent, id, init, currency);
        }

        public static PurseMoneyModel Create(IRailsMoneyOwner parent, string id, Boolean init)
        {
            Currency currency = parent.GetRoot.Bank.Currency;
            return new PurseMoneyModel(parent, id, init, currency);
        }

        new public IRailsMoneyOwner Parent
        {
            get
            {
                return (IRailsMoneyOwner)base.Parent;
            }
        }

        public Purse Purse
        {
            get
            {
                return purse;
            }
        }

        // MoneyModel abstracts
        override public int Value
        {
            get
            {
                return purse.Value();
            }
        }

        override public bool Initialized
        {
            get
            {
                return initialized.Value;
            }
        }

    }
}
